python/dicom_archive.py

```diff
@@ -0,0 +1,325 @@
+        lib.exitcode.INVALID_ARG,
+    )
+
+if (db_insert or db_update) and not profile:
+    print_error_exit(
+        'Argument \'--profile\' must be set when a \'--db-*\' argument is set.',
+        lib.exitcode.INVALID_ARG,
+    )
+
+if not os.path.isdir(source) or not os.access(source, os.R_OK):
+    print_error_exit(
+        'Argument \'--source\' must be a readable directory path.',
+        lib.exitcode.INVALID_ARG,
+    )
+
+if not os.path.isdir(target) or not os.access(target, os.W_OK):
+    print_error_exit(
+        'Argument \'--target\' must be a writable directory path.',
+        lib.exitcode.INVALID_ARG,
+    )
+
+# Connect to database (if needed)
+
+db = None
+if profile is not None:
+    db = Database(load_config_file(profile).mysql, False)
+    db.connect()
+
+# Check paths
+
+while source.endswith('/'):
+    source = source[:-1]
+
+base_name = os.path.basename(source)
+
+tar_path     = f'{target}/{base_name}.tar'
+zip_path     = f'{target}/{base_name}.tar.gz'
+summary_path = f'{target}/{base_name}.meta'
+log_path     = f'{target}/{base_name}.log'
+
+check_create_file(tar_path)
+check_create_file(zip_path)
+check_create_file(summary_path)
+check_create_file(log_path)
+
+print('Extracting DICOM information (may take a long time)')
+
+summary = lib.dicom.summary_make.make(source, verbose)
+
+if db is not None:
+    print('Checking database presence')
+
+    archive = lib.dicom.dicom_database.get_archive_with_study_uid(db, summary.info.study_uid)
+
+    if db_insert and archive is not None:
+        print_error_exit(
+            (
+                f'Study \'{summary.info.study_uid}\' is already inserted in the database\n'
+                'Previous archiving log:\n'
+                f'{archive[1]}'
+            ),
+            lib.exitcode.INSERT_FAILURE,
+        )
+
+    if db_update and archive is None:
+        print_error_exit(
+            f'No study \'{summary.info.study_uid}\' found in the database',
+            lib.exitcode.UPDATE_FAILURE,
+        )
+else:
+    # Placeholder for type checker
+    archive = None
+
+print('Copying into DICOM tar')
+
+with tarfile.open(tar_path, 'w') as tar:
+    for file in os.listdir(source):
+        tar.add(source + '/' + file)
+
+print('Calculating DICOM tar MD5 sum')
+
+tarball_md5_sum = lib.dicom.text.make_hash(tar_path, True)
+
+print('Zipping DICOM tar (may take a long time)')
+
+with open(tar_path, 'rb') as tar:
+    with gzip.open(zip_path, 'wb') as zip:
+        shutil.copyfileobj(tar, zip)
+
+print('Calculating DICOM zip MD5 sum')
+
+zipball_md5_sum = lib.dicom.text.make_hash(zip_path, True)
+
+print('Getting DICOM scan date')
+
+if not today and summary.info.scan_date is None:
+    print_warning((
+        'No scan date found for this DICOM archive, '
+        'consider using argument \'--today\' to use today\'s date instead.'
+    ))
+
+scan_date = date.today() if today else summary.info.scan_date
```

Comment:
I don't think we should use the date the script is run at a DICOM date if there is no date information found in the DICOMs. @nicolasbrossard thoughts on this?

I understand that this is to organize the DICOMs in the year subfolder but if you rerun dicomTar.pl a year later, you might end up with duplicated tar.gz file in the file system. Sounds like a recipe for confusion and duplicated files.

Reply:
I did this to match the `-today` option that currently exists in `dicomTar.pl`. I am fine with removing this option in the new script, but then it would make it impossible to insert DICOMs that do not have a scan date (unless they are patched manually to add a scan date), is this an outcome we want ?

Reply:
I added this `--today` option to match the `-today` option in the `dicomTar.pl`. But you are probably right that it might be mistake. I guess if there is no scan date we just put the archive in the root `tarchive` directory ? If the `--year` option is used in this case, should we just give a warning or abort the script execution ?